### PR TITLE
Add ShoppingCart model

### PR DESCRIPTION
**Pull request by dmarmul**
Added ShoppingCart and CartItem models, mappers, Dtos, service, repositories and controller for adding CartItem to ShoppingCart, update, delete and show items.

### CODE REVIEW

File: src/main/java/org/example/bookshop/controller/CartController.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@Tag(name = "Cart management", description = "Endpoints for managing carts")
+@RequiredArgsConstructor
+@RestController
+@RequestMapping("/cart")
+public class CartController {
+    private static final int BEARER_SUBSTRING = 7;
+
```
Comment (ahoienko) [inline src/main/java/org/example/bookshop/controller/CartController.java]:
```suggestion

```

File: src/main/java/org/example/bookshop/controller/CartController.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    private final CartService cartService;
+    private final JwtUtil jwtUtil;
+    private String userName;
+
+    @ModelAttribute
+    public void setUserName(@RequestHeader("Authorization") String token) {
+        this.userName = jwtUtil.getUsername(token.substring(BEARER_SUBSTRING));
+    }
```
Comment (ahoienko) [inline src/main/java/org/example/bookshop/controller/CartController.java]:
```suggestion
```

File: src/main/java/org/example/bookshop/controller/CartController.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    public CartDto getAll() {
+        return cartService.findAll(userName);
+    }
+
+    @PostMapping
+    @PreAuthorize("hasRole('USER')")
+    @Operation(summary = "Add books to cart",
+            description = "Add books to the shopping cart."
+                    + " Fields bookId and quantity can't must be not less than 1, can't be null.")
+    public CartItemDto add(@RequestBody @Valid CreateCartRequestDto requestDto) {
```
Comment (ahoienko) [inline src/main/java/org/example/bookshop/controller/CartController.java]:
```suggestion
    public CartItemDto add(@RequestBody @Valid CreateCartRequestDto requestDto, @AuthenticationPrincipal User user) {
```

File: src/main/java/org/example/bookshop/controller/CartController.java [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+        return cartService.add(requestDto, userName);
```
Comment (ahoienko) [inline src/main/java/org/example/bookshop/controller/CartController.java]:
```suggestion
        return cartService.add(requestDto, user);
```

File: src/main/java/org/example/bookshop/dto/CartDto.java [hunk 5]
```diff
@@ -0,0 +1,13 @@
+package org.example.bookshop.dto;
+
+import java.util.Set;
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
+@Setter
+public class CartDto {
```
Comment (ahoienko) [inline src/main/java/org/example/bookshop/dto/CartDto.java]:
```suggestion
public class ShoppingCartDto {
```

File: src/main/java/org/example/bookshop/mapper/ShoppingCartMapper.java [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+import org.example.bookshop.config.MapperConfig;
+import org.example.bookshop.dto.CartDto;
+import org.example.bookshop.mapper.impl.CartItemMapperImpl;
+import org.example.bookshop.model.ShoppingCart;
+import org.mapstruct.AfterMapping;
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.mapstruct.MappingTarget;
+
+@Mapper(config = MapperConfig.class)
```
Comment (ahoienko) [inline src/main/java/org/example/bookshop/mapper/ShoppingCartMapper.java]:
```suggestion
@Mapper(config = MapperConfig.class, uses = CartItemMapper.class)
```

File: src/main/java/org/example/bookshop/mapper/ShoppingCartMapper.java [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+    CartDto toDto(ShoppingCart cart);
+
+    @AfterMapping
+    default void setCartDtoParam(@MappingTarget CartDto cartDto, ShoppingCart cart) {
+        CartItemMapper cartItemMapper = new CartItemMapperImpl();
+        cartDto.setUserId(cart.getUser().getId());
+        cartDto.setCartItems(cart.getCartItems().stream()
+                .map(cartItemMapper::toCartItemDto)
+                .collect(Collectors.toSet()));
+    }
```
Comment (ahoienko) [inline src/main/java/org/example/bookshop/mapper/ShoppingCartMapper.java]:
```suggestion

```

File: src/main/java/org/example/bookshop/model/CartItem.java [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @ManyToOne
+    @JoinColumn(name = "shopping_cart_id")
+    private ShoppingCart shoppingCart;
+    @ManyToOne
+    private Book book;
+    @Column(nullable = false)
+    @Min(1)
```
Comment (ahoienko) [inline src/main/java/org/example/bookshop/model/CartItem.java]:
```suggestion
```

File: src/main/java/org/example/bookshop/model/CartItem.java [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+public class CartItem {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @ManyToOne
+    @JoinColumn(name = "shopping_cart_id")
+    private ShoppingCart shoppingCart;
+    @ManyToOne
+    private Book book;
+    @Column(nullable = false)
```
Comment (ahoienko) [inline src/main/java/org/example/bookshop/model/CartItem.java]: Please make fetch type Lazy

File: src/main/java/org/example/bookshop/controller/CartController.java [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+    @ModelAttribute
+    public void setUserName(@RequestHeader("Authorization") String token) {
+        this.userName = jwtUtil.getUsername(token.substring(BEARER_SUBSTRING));
+    }
+
+    @GetMapping
+    @PreAuthorize("hasRole('USER')")
+    @Operation(summary = "Get shopping cart",
+            description = "Retrieve user's shopping cart")
+    public CartDto getAll() {
```
Comment (ahoienko) [inline src/main/java/org/example/bookshop/controller/CartController.java]:
```suggestion
    public CartDto get() {
```

File: src/main/java/org/example/bookshop/service/impl/CartServiceImpl.java [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+public class CartServiceImpl implements CartService {
+    private final BookRepository bookRepository;
+    private final CartItemMapper cartItemMapper;
+    private final CartItemRepository cartItemRepository;
+    private final CartRepository cartRepository;
+    private final ShoppingCartMapper cartMapper;
+    private final UserRepository userRepository;
+
+    @Override
+    public CartDto findAll(String username) {
```
Comment (ahoienko) [inline src/main/java/org/example/bookshop/service/impl/CartServiceImpl.java]:
```suggestion
    public CartDto get(String username) {
```

File: src/main/java/org/example/bookshop/dto/CreateCartRequestDto.java [hunk 12]
```diff
@@ -0,0 +1,17 @@
+package org.example.bookshop.dto;
+
+import jakarta.validation.constraints.Min;
+import jakarta.validation.constraints.NotNull;
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
+@Setter
+public class CreateCartRequestDto {
```
Comment (ahoienko) [inline src/main/java/org/example/bookshop/dto/CreateCartRequestDto.java]:
```suggestion
public class CartItemRequestDto {
```

**Review by ahoienko — CHANGES_REQUESTED** [review]
Good job! Please check comments

File: src/main/java/org/example/bookshop/controller/CartController.java [hunk 13]
```diff
@@ -0,0 +1,68 @@
...
+    public ShoppingCartDto get(@AuthenticationPrincipal User user) {
+        return cartService.get(user);
+    }
+
+    @PostMapping
+    @PreAuthorize("hasRole('USER')")
+    @Operation(summary = "Add books to cart",
+            description = "Add books to the shopping cart."
+                    + " Fields bookId and quantity can't must be not less than 1, can't be null.")
+    public CartItemDto add(@RequestBody @Valid CartItemRequestDto requestDto,
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookshop/controller/CartController.java]:
```suggestion
    public ShoppingCartDto add(@RequestBody @Valid CartItemRequestDto requestDto,
```

File: src/main/java/org/example/bookshop/controller/CartController.java [hunk 14]
```diff
@@ -0,0 +1,68 @@
...
+                           @AuthenticationPrincipal User user) {
+        return cartService.add(requestDto, user);
+    }
+
+    @PutMapping("/items/{cartItemId}")
+    @PreAuthorize("hasRole('USER')")
+    @Operation(summary = "Update the books quantity",
+            description = "Update the books quantity in shopping cart")
+    public CartItemDto update(@RequestBody @Valid CartUpdateRequestDto requestDto,
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookshop/controller/CartController.java]:
```suggestion
    public ShoppingCartDto update(@RequestBody @Valid CartUpdateRequestDto requestDto,
```

File: src/main/java/org/example/bookshop/dto/CartItemRequestDto.java [hunk 15]
```diff
@@ -0,0 +1,17 @@
...
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
+@Setter
+public class CartItemRequestDto {
+    @NotNull
+    @Min(1)
+    private Long bookId;
+    @NotNull
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookshop/dto/CartItemRequestDto.java]: int type can't be null
```suggestion
```

File: src/main/java/org/example/bookshop/dto/CartItemRequestDto.java [hunk 16]
```diff
@@ -0,0 +1,17 @@
...
+import jakarta.validation.constraints.Min;
+import jakarta.validation.constraints.NotNull;
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
+@Setter
+public class CartItemRequestDto {
+    @NotNull
+    @Min(1)
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookshop/dto/CartItemRequestDto.java]:
```suggestion
    @Positive
```

File: src/main/java/org/example/bookshop/dto/CartItemRequestDto.java [hunk 17]
```diff
@@ -0,0 +1,17 @@
...
+    private Long bookId;
+    @NotNull
+    @Min(1)
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookshop/dto/CartItemRequestDto.java]:
```suggestion
    @Positive
```

File: src/main/java/org/example/bookshop/dto/CartUpdateRequestDto.java [hunk 18]
```diff
@@ -0,0 +1,14 @@
...
+import jakarta.validation.constraints.Min;
+import jakarta.validation.constraints.NotNull;
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
+@Setter
+public class CartUpdateRequestDto {
+    @NotNull
+    @Min(1)
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookshop/dto/CartUpdateRequestDto.java]:
```suggestion
    @Positive
```

File: src/main/java/org/example/bookshop/model/CartItem.java [hunk 19]
```diff
@@ -0,0 +1,32 @@
...
+@Getter
+@Setter
+@ToString
+@Table(name = "carts_items")
+public class CartItem {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "shopping_cart_id")
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookshop/model/CartItem.java]:
```suggestion
    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "shopping_cart_id", nullable = false)
```

File: src/main/java/org/example/bookshop/model/CartItem.java [hunk 20]
```diff
@@ -0,0 +1,32 @@
...
+    private ShoppingCart shoppingCart;
+    @ManyToOne
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookshop/model/CartItem.java]:
```suggestion
    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "book_id", nullable = false)
```

File: src/main/java/org/example/bookshop/model/CartItem.java [hunk 21]
```diff
@@ -0,0 +1,32 @@
...
+import jakarta.persistence.ManyToOne;
+import jakarta.persistence.Table;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.ToString;
+
+@Entity
+@Getter
+@Setter
+@ToString
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookshop/model/CartItem.java]:
```suggestion
```

File: src/main/java/org/example/bookshop/model/ShoppingCart.java [hunk 22]
```diff
@@ -0,0 +1,40 @@
...
+import lombok.Getter;
+import lombok.Setter;
+import lombok.ToString;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.SQLRestriction;
+
+@Entity
+@Getter
+@Setter
+@ToString
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookshop/model/ShoppingCart.java]: ToString leads to exception in such entities
```suggestion
```

File: src/main/java/org/example/bookshop/model/ShoppingCart.java [hunk 23]
```diff
@@ -0,0 +1,40 @@
...
+@SQLDelete(sql = "UPDATE shopping_carts SET is_deleted = true WHERE id=?")
+@SQLRestriction("is_deleted=false")
+@Table(name = "shopping_carts")
+public class ShoppingCart {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @OneToOne(fetch = FetchType.LAZY)
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookshop/model/ShoppingCart.java:35]:
```suggestion
    @MapsId
    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "id", nullable = false)
```

File: src/main/java/org/example/bookshop/model/ShoppingCart.java [hunk 24]
```diff
@@ -0,0 +1,40 @@
...
+    private User user;
+    @OneToMany(mappedBy = "shoppingCart")
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookshop/model/ShoppingCart.java]:
```suggestion
    @OneToMany(mappedBy = "shoppingCart", cascade = CascadeType.ALL, orphanRemoval = true)
```

File: src/main/java/org/example/bookshop/repository/CartRepository.java [hunk 25]
```diff
@@ -0,0 +1,13 @@
...
+
+import java.util.Optional;
+import org.example.bookshop.model.ShoppingCart;
+import org.springframework.data.jpa.repository.EntityGraph;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.lang.NonNull;
+
+public interface CartRepository extends JpaRepository<ShoppingCart, Long> {
+    @EntityGraph(attributePaths = {"user", "user.roles", "cartItems", "cartItems.book"})
+    @NonNull
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookshop/repository/CartRepository.java]:
```suggestion
```

File: src/main/java/org/example/bookshop/service/impl/CartServiceImpl.java [hunk 26]
```diff
@@ -0,0 +1,82 @@
...
+    private final ShoppingCartMapper cartMapper;
+
+    @Override
+    public ShoppingCartDto get(User user) {
+        ShoppingCart shoppingCart = findShoppingCart(user);
+        return cartMapper.toDto(shoppingCart);
+    }
+
+    @Override
+    public CartItemDto add(CartItemRequestDto requestDto, User user) {
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookshop/service/impl/CartServiceImpl.java]:
```suggestion
    public ShoppingCartDto add(CartItemRequestDto requestDto, User user) {
```

File: src/main/java/org/example/bookshop/service/impl/CartServiceImpl.java [hunk 27]
```diff
@@ -0,0 +1,82 @@
...
+        Book book = bookRepository.findById(requestDto.getBookId())
+                .orElseThrow(() -> new EntityNotFoundException(
+                        "Book not found with id: " + requestDto.getBookId()));
+        cartItem.setBook(book);
+
+        return cartItemMapper.toCartItemDto(cartItemRepository.save(cartItem));
+    }
+
+    @Override
+    public CartItemDto update(CartUpdateRequestDto requestDto, Long id, User user) {
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookshop/service/impl/CartServiceImpl.java]:
```suggestion
    public ShoppingCartDto update(CartUpdateRequestDto requestDto, Long id, User user) {
```

File: src/main/java/org/example/bookshop/service/impl/CartServiceImpl.java [hunk 28]
```diff
@@ -0,0 +1,82 @@
...
+            cartItemRepository.deleteById(id);
+        } else {
+            throw new EntityNotFoundException("Can't find cartItem by id: " + id);
+        }
+    }
+
+    private ShoppingCart findShoppingCart(User user) {
+        return cartRepository.findByUserId(user.getId())
+                .orElseThrow(() -> new EntityNotFoundException(
+                "Can't found cartRepository for user :" + user.getEmail())
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookshop/service/impl/CartServiceImpl.java]:
```suggestion
                "Can't find cartRepository for user :" + user.getEmail())
```

File: src/main/java/org/example/bookshop/service/impl/CartServiceImpl.java [hunk 29]
```diff
@@ -0,0 +1,82 @@
...
+    public CartItemDto update(CartUpdateRequestDto requestDto, Long id, User user) {
+        CartItem cartItem = findCartItemById(findShoppingCart(user), id);
+        cartItem.setQuantity(requestDto.getQuantity());
+
+        return cartItemMapper.toCartItemDto(cartItemRepository.save(cartItem));
+    }
+
+    @Override
+    public void delete(Long id, User user) {
+        if (findCartItemById(findShoppingCart(user), id) != null) {
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookshop/service/impl/CartServiceImpl.java]: it's not readable
find shop cart `findByUserId`
find cartItem `findByIdAndShoppingCartId`

remove cartItem

File: src/main/java/org/example/bookshop/service/impl/CartServiceImpl.java [hunk 30]
```diff
@@ -0,0 +1,82 @@
...
+import org.example.bookshop.model.User;
+import org.example.bookshop.repository.BookRepository;
+import org.example.bookshop.repository.CartItemRepository;
+import org.example.bookshop.repository.CartRepository;
+import org.example.bookshop.service.CartService;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class CartServiceImpl implements CartService {
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookshop/service/impl/CartServiceImpl.java:23]: Think where `@Transactional` is needed

File: src/main/java/org/example/bookshop/service/impl/CartServiceImpl.java [hunk 31]
```diff
@@ -0,0 +1,82 @@
...
+
+    @Override
+    public ShoppingCartDto get(User user) {
+        ShoppingCart shoppingCart = findShoppingCart(user);
+        return cartMapper.toDto(shoppingCart);
+    }
+
+    @Override
+    public CartItemDto add(CartItemRequestDto requestDto, User user) {
+        CartItem cartItem = cartItemMapper.toCartItem(requestDto);
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookshop/service/impl/CartServiceImpl.java]: it's better to find book, find shop cart, check if cart already include the same book, if yes - increment quantity, if no - add book to cart

It's better to save shop cart at the end of method

File: src/main/java/org/example/bookshop/service/impl/UserServiceImpl.java [hunk 32]
```diff
@@ -31,6 +34,11 @@ public UserResponseDto register(UserRegistrationRequestDto requestDto) {
         User user = userMapper.toModel(requestDto);
         user.setPassword(passwordEncoder.encode(user.getPassword()));
         user.setRoles(Set.of(roleRepository.findByRole(Role.RoleType.USER)));
-        return userMapper.toDto(userRepository.save(user));
+        user = userRepository.save(user);
+
+        ShoppingCart shoppingCart = new ShoppingCart();
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookshop/service/impl/UserServiceImpl.java]: create method `createShoppingCart(User user)` in ShopCartService and implement the logic

File: src/main/java/org/example/bookshop/service/impl/UserServiceImpl.java [hunk 33]
```diff
@@ -31,6 +34,11 @@ public UserResponseDto register(UserRegistrationRequestDto requestDto) {
         User user = userMapper.toModel(requestDto);
         user.setPassword(passwordEncoder.encode(user.getPassword()));
         user.setRoles(Set.of(roleRepository.findByRole(Role.RoleType.USER)));
-        return userMapper.toDto(userRepository.save(user));
+        user = userRepository.save(user);
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookshop/service/impl/UserServiceImpl.java]:
```suggestion
        userRepository.save(user);
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

File: src/main/java/org/example/bookshop/service/impl/UserServiceImpl.java [hunk 34]
```diff
@@ -31,6 +33,7 @@ public UserResponseDto register(UserRegistrationRequestDto requestDto) {
         User user = userMapper.toModel(requestDto);
         user.setPassword(passwordEncoder.encode(user.getPassword()));
         user.setRoles(Set.of(roleRepository.findByRole(Role.RoleType.USER)));
-        return userMapper.toDto(userRepository.save(user));
+        cartService.createShoppingCart(userRepository.save(user));
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookshop/service/impl/UserServiceImpl.java]:
```suggestion
        userRepository.save(user)
        cartService.createShoppingCart(user);
```

File: src/main/java/org/example/bookshop/service/impl/CartServiceImpl.java [hunk 35]
```diff
@@ -0,0 +1,109 @@
...
+    public void delete(Long id, User user) {
+        if (cartItemRepository.findByIdAndUserId(id, user.getId()).isPresent()) {
+            cartItemRepository.deleteById(id);
+        } else {
+            throw new EntityNotFoundException("Can't find cartItem by id: " + id);
+        }
+    }
+
+    @Override
+    @Transactional
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookshop/service/impl/CartServiceImpl.java]:
```suggestion
```

File: src/main/java/org/example/bookshop/service/impl/CartServiceImpl.java [hunk 36]
```diff
@@ -0,0 +1,109 @@
...
+    public void createShoppingCart(User user) {
+        ShoppingCart shoppingCart = new ShoppingCart();
+        shoppingCart.setUser(userRepository.findById(user.getId()).orElseThrow(() ->
+                new EntityNotFoundException("Not found user with id: " + user.getId())
+        ));
+        cartRepository.save(shoppingCart);
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookshop/service/impl/CartServiceImpl.java]:
```suggestion
        shoppingCart.setUser(user);
        cartRepository.save(shoppingCart);
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

File: src/main/java/org/example/bookshop/dto/CartItemDto.java [hunk 37]
```diff
@@ -0,0 +1,13 @@
+package org.example.bookshop.dto;
+
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
+@Setter
```
Comment (okuzan) [inline src/main/java/org/example/bookshop/dto/CartItemDto.java:7]: Not a mistake but FYI it's ok to use @Data from lombok on DTOs, just not JPA entities

File: src/main/java/org/example/bookshop/service/impl/CartServiceImpl.java [hunk 38]
```diff
@@ -0,0 +1,104 @@
...
+        return shoppingCartMapper.toDto(cartRepository.save(shoppingCart));
+    }
+
+    @Override
+    public void delete(Long id, User user) {
+        if (cartItemRepository.findByIdAndUserId(id, user.getId()).isPresent()) {
+            cartItemRepository.deleteById(id);
+        } else {
+            throw new EntityNotFoundException("Can't find cartItem by id: " + id);
+        }
```
Comment (okuzan) [inline src/main/java/org/example/bookshop/service/impl/CartServiceImpl.java]:
```suggestion
        if (cartItemRepository.findByIdAndUserId(id, user.getId()).isEmpty()) {
            throw new EntityNotFoundException("Can't find cartItem by id: " + id);
        }
        cartItemRepository.deleteById(id);
```

Comment (okuzan) [reply src/main/java/org/example/bookshop/service/impl/CartServiceImpl.java]: we don't need else after returning throwing, so in such cases, to simplify the code, invert if condition

**Review by okuzan — APPROVED** [review]

**Pull request merged**